src/hidimstat/distilled_conditional_randomization_test.py

```diff
@@ -499,38 +460,34 @@ def _joblib_fit(
     # Distill X with least square loss
     # configure Lasso and determine the alpha
     if sigma_X:
-        clf_x = _fit_lasso(
-            X_minus_idx,
-            X[:, idx],
-            n_jobs=1,
-            **params_lasso_distillation_x,
-        )
+        model_x = clone(model_distillation_x)
+        model_x.fit(X_minus_idx, X[:, idx])
     else:
-        clf_x = None
+        model_x = None
 
     if (isinstance(estimator, Lasso) or isinstance(estimator, LassoCV)) and not fit_y:
```

Comment:
I think we need to check if it's a linear model instead than if it's lasso.
What do you think?